tests/integrations/test_celery.py

```diff
@@ -0,0 +1,192 @@
+from contextlib import contextmanager
+import pytest
+import celery
+from celery import shared_task
+from celery.signals import task_failure
+from bugsnag.celery import connect_failure_handler, failure_handler
+from tests.utils import MissingRequestError
+
+
+@pytest.fixture(scope='function')
+def celery_config():
+    return {
+        'broker_url': 'memory://',
+        'result_backend': 'rpc',
+    }
+
+
+@contextmanager
+def celery_failure_handler():
+    """
+    The bugsnag celery integration works by listening to the celery
+    task_failure signal and sending an event when the signal is received.
+
+    This context manages the signal connection and ensures that error handling
+    does not occur across separate tests.
+    """
+    connect_failure_handler()
+    try:
+        yield
+        task_failure.disconnect(failure_handler)
+    except Exception:
+        task_failure.disconnect(failure_handler)
+        raise
+
+
+def test_app_task_operation(celery_app, celery_worker, bugsnag_server):
+    """
+    Configuring bugsnag should not interfere with tasks succeeding normally
+    """
+
+    @celery_app.task
+    def square(x):
+        return x * x
+
+    celery_worker.reload()
+
+    with celery_failure_handler():
+        assert square.delay(3).get(timeout=1) == 9
+
+    try:
+        bugsnag_server.wait_for_request()
+    except MissingRequestError:
+        pass
```

Comment:
I'm not sure I'm convinced, but maybe this is clearer using `pytest.raises` to indicate that we expect an exception?

```suggestion
    with pytest.raises(MissingRequestError):
        bugsnag_server.wait_for_request()
```

also on line 152